staging/src/k8s.io/dynamic-resource-allocation/kubeletplugin/nonblockinggrpcserver.go

```diff
@@ -78,13 +78,17 @@ func startGRPCServer(logger klog.Logger, grpcVerbosity int, unaryInterceptors []
 		defer s.wg.Done()
 		err := s.server.Serve(listener)
 		if err != nil {
-			logger.Error(err, "GRPC server failed")
+			if cancelMainContext != nil {
+				cancelMainContext(fmt.Errorf("gRPC server failed to serve on %s: %w", endpoint.path(), err))
+			} else {
+				logger.Error(err, "gRPC server failed to serve", "endpoint", endpoint.path())
+			}
```

Comment:
why a grpc server error should cancel a context that is up in the chain of context cancellation?